cmd/archive_upload_client/client.go

```diff
@@ -62,8 +63,16 @@ func upload(ctx context.Context, conn *grpc.ClientConn, p *pb.FileRequest, audie
 	// A given TokenSource is specific to the audience.
 	tokenSource, err := idtoken.NewTokenSource(ctx, audience)
 	if err != nil {
```

Comment:
I was trying to run the code in this PR on a fresh machine (no gcloud, no login credentials) and it only returns me 401 unauthorized, even though I've set GOOGLE_APPLICATION_CREDENTIALS and granted correct permissions to the service account.

After some investigation, it turns out that if we add a grpc.WithPerRPCCredentials to the gRPC options, the issue is gone. I don't really know the difference between WithPerRPCCredentials and metadata.AppendToOutgoingContext, but it just seems to work... Perhaps the credentials comes earlier then the metadata in a gRPC request? 

To make this work for any fresh machine, a few changes will be needed:
1. Generate a token and add the credentials using grpc.WithPerRPCCredentials to the dial options.
2. Remove token & metadata from upload().
See [this](https://github.com/routeviews/google-cloud-storage/commit/8123220c8cc1872e4ffe2d9154b5b5ef150c1d5e) for example.

Ref: https://github.com/salrashid123/cloud_run_grpc_auth

Reply:
I'm happy to rm-rf this PR.. and take up your option :)
I'll delete the branch and clsoe the pr here.

Reply:
One comment on your PR, we should update/add documentation for the client to say how to invoke it, what's required to get to 'sends files fine'.